pkg/controller/core/workload_controller.go

```diff
@@ -293,6 +301,57 @@ func isDisabledRequeuedByClusterQueueStopped(w *kueue.Workload) bool {
 	return cond != nil && cond.Status == metav1.ConditionFalse && cond.Reason == kueue.WorkloadEvictedByClusterQueueStopped
 }
 
+func (r *WorkloadReconciler) isScaledDown(wl *kueue.Workload) bool {
+	if !features.Enabled(features.ResizableJobs) {
+		return false
+	}
+	podSetSize := len(wl.Spec.PodSets)
+	for i := 1; i < podSetSize; i++ {
+		if ptr.Deref(wl.Status.Admission.PodSetAssignments[i].Count, 0) > wl.Spec.PodSets[i].Count {
+			return true
+		}
+	}
+	return false
+}
+
+func (r *WorkloadReconciler) downSizeJobIfNecessary(wl *kueue.Workload, ctx context.Context) error {
+	updateStatus := false
+	podSetSize := len(wl.Spec.PodSets)
+	for i := 1; i < podSetSize; i++ {
+		if ptr.Deref(wl.Status.Admission.PodSetAssignments[i].Count, 0) <= wl.Spec.PodSets[i].Count {
+			continue
+		}
+		// Get Resource Requests and Usage values
+		originalResourceRequests := limitrange.TotalRequests(&wl.Spec.PodSets[i].Template.Spec)
+		currentAssignedResourceUsage := wl.Status.Admission.PodSetAssignments[i].ResourceUsage
+
+		diff := ptr.Deref(wl.Status.Admission.PodSetAssignments[i].Count, 0) - wl.Spec.PodSets[i].Count
+		for k := range currentAssignedResourceUsage {
+			resourceQuantity := originalResourceRequests[k]
+			resourceQuantity.Mul(int64(diff))
+			originalResourceRequests[k] = resourceQuantity
```

Comment:
I suppose not change the original resources, although will not impact the result but may lead to confusion?